persistence/src/main/java/org/mskcc/cmo/metadb/persistence/SampleMetadataRepository.java

```diff
@@ -0,0 +1,46 @@
+package org.mskcc.cmo.metadb.persistence;
+
+import org.mskcc.cmo.messaging.model.SampleMetadataEntity;
+
+import java.util.List;
+import java.util.UUID;
+import org.springframework.data.neo4j.annotation.Query;
+import org.springframework.data.neo4j.repository.Neo4jRepository;
+import org.springframework.data.repository.query.Param;
+
+/**
+ *
+ * @author ochoaa
+ */
+public interface SampleMetadataRepository extends Neo4jRepository<SampleMetadataEntity, UUID> {
+    @Query("MATCH (s:SampleMetadata) RETURN DISTINCT(s.sampleName)")
+    List<String> findAllSampleNames();
+
+    @Query("MATCH (s:SampleMetadata) WHERE $investigatorSampleId = s.investigatorSampleId RETURN s")
+    SampleMetadataEntity findSampleByInvestigatorId(@Param("investigatorSampleId") String investigatorSampleId);
+
+    /**
+     * TODO: Follow up with LIMS/IGO folks.
+     *
+     *  - Will IGO sample ID be the same for a given sample even if it is updated in LIMS?
+     *  - In the event that we get an update for an existing sample in the metadb, will all of the properties for that sample be published or will only the updated properties be sent?
+     * @param sample
+     */
+    @Query(
+        "MERGE (sm:SampleMetadata {investigatorSampleId: $sample.investigatorSampleId}) " +
```

Comment:
We should discuss this - I think that the cmo_metadb_sample_id (uuid) should be used as a primary key for metadata update.  After the node is created for the first time, we should be sharing the uuid to external systems for future reference to the node.